### ISSUE TITLE
MySQL ERROR 1045 (28000): Access denied for user 'root'@'localhost' ...FATAL ERROR: Upgrade failed

### ISSUE THREAD

**Issue by holta** [iiab/iiab#2475]
Is this very ugly series of errors in `/var/log/syslog` fixable?  Even if some suggest it's possibly inconsequential?

```
Aug  5 16:18:39 box /etc/mysql/debian-start[1064]: Upgrading MySQL tables if necessary.
Aug  5 16:18:39 box /etc/mysql/debian-start[1070]: /usr/bin/mysql_upgrade: the '--basedir' option is always ignored
Aug  5 16:18:40 box /etc/mysql/debian-start[1070]: Looking for 'mysql' as: /usr/bin/mysql
Aug  5 16:18:40 box /etc/mysql/debian-start[1070]: Looking for 'mysqlcheck' as: /usr/bin/mysqlcheck
Aug  5 16:18:40 box /etc/mysql/debian-start[1070]: Version check failed. Got the following error when calling the 'mysql' command line client
Aug  5 16:18:40 box /etc/mysql/debian-start[1070]: ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: [REDACTED]
Aug  5 16:18:40 box /etc/mysql/debian-start[1070]: FATAL ERROR: Upgrade failed
Aug  5 16:18:40 box /etc/mysql/debian-start[1095]: Checking for insecure root accounts.
```

Excerpted from short-term pastebin https://paste.ubuntu.com/p/J7f7SG3vdW/

(But this seemingly affects all installs onto Raspberry Pi OS...and possibly other/all installs onto Debian-related distros ...as suggested by debian-start in the error messages ??)

Ref:

#1714 'MySQL database password-changing failure during fresh install of IIAB [using "sudo iiab"] on Ubuntu 19.04 [and now 18.04 too...whereas 19.10 & Debian 10.x work!] -- Ansible Python 3 needs python3-distutils'

**Comment by holta** [comment]
While oddly not catastrophic to IIAB's general functioning, this is a longstanding issue @jvonau has seen for many months.

He & @tim-moody suspect the password is being stored in the wrong file / wrong place (i.e. in a different place from where the service is expecting it).

Likely solvable by copying the password/credential into the right place?!

Likely Related:

- /root/.my.cnf
- /etc/mysql/my.cnf -> /etc/alternatives/my.cnf
- /etc/mysql/my.cnf.fallback
- /etc/mysql/mariadb.cnf

**Comment by holta** [comment]
Worse, this my.cnf file might need to be under `/home/iiab-admin` or `/home/pi` (or similar) depending on...what user ran 'sudo iiab' to install IIAB?

**Comment by holta** [comment]
Thanks to @jvonau working hard on a (likely) fix here: PR #2488

**Comment by holta** [comment]
Can this be closed now?

**Comment by holta** [comment]
This error on boot seems solved by @jvonau's PR #2488 now merged.

**Comment by holta** [comment]
Partially related followup:

#2489 "FreePBX fixes for php7.4 on Ubuntu 20.04"
